swarm/network/stream/stream.go

```diff
@@ -509,6 +492,69 @@ func (r *Registry) updateSyncing() {
 	}
 }
 
+// requestPeerSubscriptions calls on each live peer in the kademlia table
+// and sends a `RequestSubscription` to peers according to their bin
+// and their relationship with kademlia's depth.
+// Also check `TestRequestPeerSubscriptions` in order to understand the
+// expected behavior.
+// The function expects:
+//   * the kademlia
+//   * a map of subscriptions
+//   * the actual function to subscribe
+//     (in case of the test, it doesn't do real subscriptions)
+func (r *Registry) requestPeerSubscriptions(
+	kad *network.Kademlia,
+	subs map[enode.ID]map[Stream]struct{},
```

Comment:
Why aren't the subs part of the `Registry` object?

(Although the `Registry` object _does_ seem a bit crowded already)

Reply:
See comment above, @zelig ?

Reply:
Not addressed. So you decided against?

Reply:
As far as I can see, `subs` are manipulated only in the context of the `updateSyncing` function - they have a "volatile" nature. Thus, it is justified to not have them as a member in `Registry`. 

But I don't have  a  strong opinion on this. @zelig ?

Reply:
i agree

Reply:
ok